mlrun/utils/regex.py

```diff
@@ -70,4 +70,8 @@
 # of the 3 restrictions, dns 1123 label is the most strict, so we enforce only it
 project_name = dns_1123_label
 
+# Special characters are not permitted in tag names because they can be included in the url and cause problems.
+# We only accept letters, capital letters, numbers, dots, and hyphens, with a character limit of 63.
+tag_name = [r"^.{0,63}$", r"^(([A-Za-z0-9][-A-Za-z0-9.]*)?[A-Za-z0-9])?$"]
+
```

Comment:
The first validation seems to be used in multiple regex validations, I would create new validation name called `k8s_character_limit = [r"^.{0,63}$"]` and use it in all the relevant validations.